features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemImageView.kt

```diff
@@ -16,39 +16,126 @@
 import androidx.compose.ui.semantics.contentDescription
 import androidx.compose.ui.semantics.semantics
 import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
+import coil.compose.AsyncImagePainter
+import io.element.android.features.messages.impl.timeline.aTimelineItemEvent
+import io.element.android.features.messages.impl.timeline.components.ATimelineItemEventRow
+import io.element.android.features.messages.impl.timeline.components.layout.ContentAvoidingLayout
+import io.element.android.features.messages.impl.timeline.components.layout.ContentAvoidingLayoutData
+import io.element.android.features.messages.impl.timeline.model.TimelineItemGroupPosition
 import io.element.android.features.messages.impl.timeline.model.event.TimelineItemImageContent
 import io.element.android.features.messages.impl.timeline.model.event.TimelineItemImageContentProvider
-import io.element.android.libraries.designsystem.components.BlurHashAsyncImage
+import io.element.android.features.messages.impl.timeline.model.event.aTimelineItemImageContent
+import io.element.android.libraries.designsystem.components.blurhash.blurHashBackground
 import io.element.android.libraries.designsystem.preview.ElementPreview
 import io.element.android.libraries.designsystem.preview.PreviewsDayNight
+import io.element.android.libraries.matrix.api.timeline.item.event.MessageFormat
 import io.element.android.libraries.matrix.ui.media.MediaRequestData
+import io.element.android.libraries.textcomposer.ElementRichTextEditorStyle
 import io.element.android.libraries.ui.strings.CommonStrings
+import io.element.android.wysiwyg.compose.EditorStyledText
 
 @Composable
 fun TimelineItemImageView(
     content: TimelineItemImageContent,
+    onContentLayoutChanged: (ContentAvoidingLayoutData) -> Unit,
     modifier: Modifier = Modifier,
 ) {
     val description = stringResource(CommonStrings.common_image)
-    TimelineItemAspectRatioBox(
-        aspectRatio = content.aspectRatio,
+    Column(
         modifier = modifier.semantics { contentDescription = description },
     ) {
-        BlurHashAsyncImage(
-            model = MediaRequestData(content.preferredMediaSource, MediaRequestData.Kind.File(content.body, content.mimeType)),
-            blurHash = content.blurhash,
-        )
+        val containerModifier = if (content.showCaption) {
+            Modifier.padding(top = 6.dp).clip(RoundedCornerShape(6.dp))
+        } else {
+            Modifier
+        }
+        TimelineItemAspectRatioBox(
+            modifier = containerModifier.blurHashBackground(content.blurhash, alpha = 0.9f),
+            aspectRatio = content.aspectRatio,
+        ) {
+            var isLoaded by remember { mutableStateOf(false) }
+            AsyncImage(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .then(if (isLoaded) Modifier.background(Color.White) else Modifier),
```

Comment:
> FTR last time I checked, Element Web does not add white BG to images.

Yes, but Matthew said it was a mistake. i.e. png images with no background and black letters were impossible to read in dark mode.

> Is the white background the cause of these strange artifacts on the border/corners?

Sadly, I think that's the case...

Reply:
> png images with no background and black letters

OK, so with a white background, how a png image with no background and white letters will render?

I'd say this is the problem for only a minority of files, probably built just to make applications fail (not only Element).

Adding a background to image with transparency is like not supporting image with transparency, this is a shame. I still believe we should not add a white background. This degrades the user experience (stickers, etc.)

Reply:
You might be right. If the users find it annoying we can always remove this white background.

> This degrades the user experience (stickers, etc.)

Stickers use a different view, so they won't have this problem.

Reply:
> Yes, but Matthew said it was a mistake. i.e. png images with no background and black letters were impossible to read in dark mode.

Could we only add the background if we detect that the image is both transparent, and has a specific contrast ratio relative to the app colour scheme?

Reply:
To check if an image is transparent we'd have to load it twice in memory if I'm not mistaken, which isn't really efficient, or add some transformation when Coil loads them, but it seems like it could affect animated images (GIFs, WebP...), so I don't think it's something we want to invest too much time on.